packages/core/src/providers/user/thea/index.tsx

```diff
@@ -0,0 +1,373 @@
+  const destinationConnector = useMemo(
+    () => destinationChain && getChainConnector(destinationChain.genesis),
+    [destinationChain]
+  );
+
+  const polkadexConnector = useMemo(
+    () => destinationChain && getChainConnector(GENESIS[0]),
+    [destinationChain]
+  );
+
+  const sourceAssets = useMemo(
+    () => (sourceConnector ? sourceConnector.getSupportedAssets() : []),
+    [sourceConnector]
+  );
+
+  const polkadexAssets = useMemo(
+    () => (polkadexConnector ? polkadexConnector.getSupportedAssets() : []),
+    [polkadexConnector]
+  );
+
+  const destinationAssets = useMemo(
+    () =>
+      destinationConnector ? destinationConnector.getSupportedAssets() : [],
+    [destinationConnector]
+  );
+
+  const supportedAssets = useMemo(
+    () =>
+      sourceAssets && destinationAssets
+        ? sourceAssets?.filter((e) =>
+            destinationAssets?.some((x) => e?.ticker === x?.ticker)
+          )
+        : [],
+    [sourceAssets, destinationAssets]
+  );
+
+  const initialAsset = useMemo(() => {
+    if (supportedAssets) {
+      const asset =
+        initialAssetTicker &&
+        isIdentical(supportedAssets, initialAssetTicker, "ticker");
+      return asset || supportedAssets[0];
+    }
+  }, [initialAssetTicker, supportedAssets]);
+
+  const initialSource = useMemo(() => {
+    if (chains) {
+      return !!initialSourceName && initialSourceName !== initialDestinationName
+        ? isIdentical(chains, initialSourceName, "name")
+        : isIdentical(chains, defaultTheaSourceChain, "name");
+    }
+  }, [initialSourceName, chains, initialDestinationName]);
+
+  const initialDestination = useMemo(() => {
+    if (chains) {
+      return !!initialDestinationName &&
+        initialSourceName !== initialDestinationName
+        ? isIdentical(chains, initialDestinationName, "name")
+        : isIdentical(chains, defaultTheaDestinationChain, "name");
+    }
+  }, [initialDestinationName, initialSourceName, chains]);
+
+  useEffect(() => {
+    if (initialDestination && !destinationChain)
+      setDestinationChain(initialDestination);
+  }, [initialDestination, destinationChain]);
+
+  useEffect(() => {
+    if (initialSource && !sourceChain) setSourceChain(initialSource);
+  }, [initialSource, sourceChain]);
+
+  useEffect(() => {
+    if (!selectedAsset && destinationChain && sourceChain && supportedAssets)
+      setSelectedAsset(initialAsset || supportedAssets[0]);
+  }, [
+    initialAsset,
+    selectedAsset,
+    destinationChain,
+    sourceChain,
+    supportedAssets,
+  ]);
+
+  const {
+    data: sourceBalances = [],
+    isLoading: sourceBalancesLoading,
+    isFetching: sourceBalancesFetching,
+    isSuccess: sourceBalancesSuccess,
+    refetch: sourceBalancesRefetch,
+  } = useTheaBalances({
+    connector: sourceConnector,
+    sourceAddress: sourceAccountSelected?.address,
+    assets: sourceAssets,
+    chain: sourceChain?.genesis,
+  });
+
+  const onRefetchSourceBalances = useCallback(async () => {
```

Comment:
Is this function really required ? It's just calling `sourceBalancesRefetch` in it. I think, we should directly return it like this - 

```js
return { 
....
onRefetchSourceBalances: sourceBalancesRefetch 
....
}

```

Reply:
fixed

Reply:
This basically avoided typing this function. I have corrected it as an optional function

![Screenshot 2024-05-20 at 06 53 52](https://github.com/Polkadex-Substrate/Polkadex-Orderbook-Frontend/assets/12574469/6a18ab0c-473a-45c2-a57d-053869576dd2)